### PR TITLE
PPTC Follow-up.

### PR DESCRIPTION
**Pull request by LDj3SNuD**
- Get rid of `BinaryFormatter` (is observable a reduction of the size of both the .info and the .cache files and a faster deserialization/serialization (it implies reduced loading/saving times));
- Added invalidation/stubbing functionality of rejited functions, between runs, at .cache file level (is observable a reduction of its size and a saving of JitCache memory);
- NativeInterface.GetIndirectFunctionAddress(): Validated that writing actually takes place in dynamic table memory range (and not elsewhere);
- Added header with magic and file version for the internal (uncompressed) stream of .info files (as .cache files do);
- Added .info and .cache files invalidation in case of a failed stream decompression (addressed in https://github.com/Ryujinx/Ryujinx/pull/1704);
- Added invalidation of .cache files in the event of reuse on a different user operating system;
- PPTC loading no longer blocks the GUI (addressed in https://github.com/Ryujinx/Ryujinx/pull/1518);
- Logs:
  - Added info/cache Load/Save logs;
  - Added LoadTranslations log;
  - Improved MakeAndSaveTranslations logs.

Nits.

---

***Note for users:** Pre-existing .info (profiling) files will be invalidated.*

### CODE REVIEW

**Comment by Thog** [comment]
Rebased required because of https://github.com/Ryujinx/Ryujinx/commit/aa129fdbdfd424f73da2b187fc28855f8ae90059.

**Comment by LDj3SNuD** [comment]
Rebased due to .NET 5.0.
2 reviewers required.

File: ARMeilleure/Translation/PTC/Ptc.cs [hunk 1]
```diff
@@ -663,12 +782,25 @@ private static ulong GetFeatureInfo()
             return featureInfo;
         }
 
+        private static uint GetOSPlatform()
+        {
+            uint oSPlatform = 0u;
```
Comment (gdkchan) [inline ARMeilleure/Translation/PTC/Ptc.cs:787]: Maybe call it `osPlatform`?

**Comment by LDj3SNuD** [comment]
Ready to be reviewed.

**Comment by LDj3SNuD** [comment]
Reminder: reviews required.

**Review by Thog — APPROVED** [review]
lgtm!

**Comment by LDj3SNuD** [comment]
Reminder: review required.

**Comment by LDj3SNuD** [comment]
Ready to be reviewed.

**Comment by gdkchan** [comment]
> Compatibility with #1518 preserved (borrowed part of the code, which PR will be merged first it is possible to rebase the other one easily).

Feel free to take all the fixes, I can just close it then, sounds like that might be easier.
Needs a rebase btw.

**Review by gdkchan — APPROVED** [review]
lgtm, thanks

**Comment by LDj3SNuD** [comment]
> Feel free to take all the fixes, I can just close it then, sounds like that might be easier.

Why close your PR (I don't need any more code from it). If you review my proposal about JumpTable.cs https://github.com/Ryujinx/Ryujinx/pull/1518#issuecomment-735340052, I can rebase your PR too. It's up to you to decide which one to merge first.

**Comment by LDj3SNuD** [comment]
Rebased and ready to be reviewed/merged.

**Pull request merged**